docs/proposal/architecture_for_diff_iaas.md

```diff
@@ -0,0 +1,32 @@
+# IBM Cloud Provider with multiple types of IaaS
+
+## Motivation
+
+There are multiple types of IaaS in IBM Cloud: Classic infrastructure, VPC and PowerVS etc. Each type of IaaS has some specific features. Users would like to bootstrap a k8s cluster from any types of IaaS. It's hard to maintain an individual repository for each type of IaaS. A good implementation should describe IBM Cloud feature in an appropriate grain. 
+
+## Proposal
+
+This proposal presents a 2 layers infrastructure resource definition. Th top layer describes IBM Cloud common feature and the 2nd layer describes the details of the specific IaaS implementation.
+
+![](../images/cluster-api-ibmcloud-definition.png)
+
```

Comment:
I assume one cluster-api-ibmcloud pod will watch all of the resources and in one cluster we can handle all those 3 resources?

Reply:
Yes